### DISCUSSION TITLE
Change container network configuration

### DISCUSSION THREAD

**Discussion by assistenzasistemisticadab** [telekom-security/tpotce#978] [Q&A]
Hi,
I would like to know how I can change the network configuration of one or more containers before they start (I can change it from the gui after they start).
What are the configuration files from which these settings are taken ?
Thanks

**Comment by t3chn0m4g3** [comment]
You have to adjust your `/opt/tpot/etc/tpot.yml` to your needs. The **[docker-compose](https://docs.docker.com/compose/compose-file/compose-file-v2/)** online reference should have you covered.

**Reply by assistenzasistemisticadab** [reply]
Ok, thank you for the support... but I can't figure out where these networks are defined:

docker network list

NETWORK ID          NAME                            DRIVER              SCOPE
d09be9e2b359        bridge                          bridge              local
6cbc826e6ce3        etc_adbhoney_local              bridge              local
0536df537129        etc_citrixhoneypot_local        bridge              local
3d3af276b322        etc_conpot_local_IEC104         bridge              local
3ff11d561b8e        etc_conpot_local_guardian_ast   bridge              local
8b22424a3c13        etc_conpot_local_ipmi           bridge              local
05cea691a095        etc_conpot_local_kamstrup_382   bridge              local
de5ff1cfddb8        etc_cowrie_local                bridge              local
a87c05fc64a1        etc_cyberchef_local             bridge              local
dfd364510392        etc_default                     bridge              local
dc0958cc3c1d        etc_dicompot_local              bridge              local
14800ce80a33        etc_dionaea_local               bridge              local
25e22dcf747c        etc_ewsposter_local             bridge              local
4aae54b44992        etc_heralding_local             bridge              local
09c3c80854db        etc_honeysap_local              bridge              local
8f9708652efe        etc_mailoney_local              bridge              local
a8d304e8438f        etc_medpot_local                bridge              local
bf0e57fd2e60        etc_rdpy_local                  bridge              local
f3667c53fa5b        etc_spiderfoot_local            bridge              local
90a25c30436a        etc_tanner_local                bridge              local
445bb2f466c7        host                            host                local
2d1730d808b7        none                            null                local

I need to modify etc_rdpy_local from 192.168.0.0/20 to 192.168.0.0/24 

Thank you

**Comment by cl4x** [comment]
We have solved with this bash script in cron:

```
#!/bin/bash
ip addr show | grep "br-" | awk -F"br-" '{ print $2 }' | grep "state" | awk '{print $1}' | grep ":" | awk -F":" '{print $1}' > /root/tmp_changenet

TOTNET=$( cat /root/tmp_changenet | wc -l )
COUNTNET=1

while [ $COUNTNET -le $TOTNET ]; do
        ACTINT=$( cat /root/tmp_changenet | sed -n $COUNTNET\p );
        ACTINTFULL=br-$ACTINT;
        ACTNET=$( ip addr show $ACTINTFULL | grep inet | awk '{ print $2 }' );
        echo "Interfaccia $ACTINTFULL, NET: $ACTNET";

        if [ "$ACTNET" == "192.168.16.1/20" ]; then
                ip addr del 192.168.16.1/20 dev $ACTINTFULL;
                ip addr add 192.168.16.1/24 dev $ACTINTFULL;
                echo "Cambiata network!";
        fi;

        if [ "$ACTNET" == "192.168.0.1/20" ]; then
                ip addr del 192.168.0.1/20 dev $ACTINTFULL;
                ip addr add 192.168.0.1/24 dev $ACTINTFULL;
                echo "Cambiata network!";
        fi;

#ip addr del 192.168.16.1/20 dev br-3b4e3e37dd4a
#ip addr add 192.168.16.1/24 dev br-3b4e3e37dd4a
#ip addr del 192.168.0.1/20 dev br-c82bb50c3227
#ip addr add 192.168.0.1/24 dev br-c82bb50c3227


        let COUNTNET=COUNTNET+1
done
```




in crontab:

```
00,05,10,15,20,25,30,35,40,45,50,55 * * * * /root/changenet.sh
@reboot /root/changenet.sh
```



OUTPUT:
![image](https://user-images.githubusercontent.com/47188814/149086092-f62a2dd8-6da5-41ab-9735-c050f4cc9ad7.png)


if network are 192.168.0.1/20 or 192.168.16.1/20, it change in /24.
